axum/src/extract/host.rs

```diff
@@ -0,0 +1,64 @@
+use super::{FromRequest, RequestParts, rejection::{HostRejection, FailedToResolveHost}};
+use async_trait::async_trait;
+
+/// Extractor that extracts the host from a request.
```

Comment:
I think we should document how we try to resolve the host, and in which order. [Similarly to what actix does.](https://docs.rs/actix-web/latest/actix_web/dev/struct.ConnectionInfo.html#method.host)